assets/src/edit-story/components/form/radioGroup/index.js

```diff
@@ -13,102 +13,75 @@
 import useRadioNavigation from '../shared/useRadioNavigation';
 
-const Selected = styled(RadioSelected)`
-  color: ${({ theme }) => theme.DEPRECATED_THEME.colors.radio};
-`;
-
-const RadioButton = styled.label`
-  display: block;
-  min-height: 30px;
-  margin: 10px 0;
-`;
-
-const RADIO_SIZE = 24;
-const TEXT_OFFSET = 30;
-
-const Label = styled.span`
-  display: flex;
-  position: relative;
-  margin-bottom: 0;
-  cursor: pointer;
-  svg {
-    width: ${RADIO_SIZE}px;
-    height: ${RADIO_SIZE}px;
-    margin-right: ${TEXT_OFFSET - RADIO_SIZE}px;
-  }
+const VisuallyHiddenLabel = styled(Headline)`
+  ${themeHelpers.visuallyHidden};
 `;
 
-const Name = styled.span`
-  line-height: 24px;
-`;
-
-// Class should contain "mousetrap" to enable keyboard shortcuts on inputs.
-const Radio = styled.input.attrs({ className: 'mousetrap' })`
-  opacity: 0;
-  position: absolute;
-  ${KEYBOARD_USER_SELECTOR} &:focus + ${Label} {
-    outline: -webkit-focus-ring-color auto 5px;
-  }
-`;
-
-const Helper = styled.div`
-  margin-left: ${TEXT_OFFSET}px;
-  color: ${({ theme }) => rgba(theme.DEPRECATED_THEME.colors.fg.white, 0.54)};
-  font-size: 12px;
-  line-height: ${({ theme }) => theme.DEPRECATED_THEME.fonts.label.lineHeight};
-`;
+function RadioGroup({
+  className,
+  groupLabel,
+  id,
+  name,
+  options,
+  value: selectedValue,
+  ...radioButtonProps
+}) {
+  const groupId = useMemo(() => uuidv4(), []);
+  const groupRef = useRef(null);
+  useRadioNavigation(groupRef);
 
-function RadioGroup({ onChange, value: selectedValue, options }) {
-  const radioGroupId = useMemo(() => uuidv4(), []);
-
-  // We need manual arrow key navigation here, as we have a global listener for those keys
-  // preventing default functionality.
-  const ref = useRef();
-
-  useRadioNavigation(ref);
   return (
-    <div ref={ref}>
-      {options.map(({ value, name, helper }, i) => (
-        <RadioButton key={value}>
-          <Radio
-            onChange={(evt) => onChange(evt.target.value, evt)}
-            value={value}
-            type="radio"
-            checked={value === selectedValue}
-            aria-labelledby={`radio-${i}-${radioGroupId}`}
-          />
-          <Label isActive={value === selectedValue}>
-            {value === selectedValue ? <Selected /> : <Unselected />}
-            <Name id={`radio-${i}-${radioGroupId}`}>{name}</Name>
-          </Label>
-          {helper && <Helper>{helper}</Helper>}
-        </RadioButton>
+    <div
+      className={className}
+      ref={groupRef}
+      role="radiogroup"
+      aria-labelledby={groupId}
+    >
+      <VisuallyHiddenLabel id={groupId} as="h4">
+        {groupLabel}
+      </VisuallyHiddenLabel>
+      {options.map(({ helper, label, value }) => (
+        <Radio
+          key={value}
+          name={name}
+          value={value}
+          checked={value === selectedValue}
+          label={label}
+          hint={helper}
+          {...radioButtonProps}
+        />
```

Comment:
Decided not to add the `RadioGroup` to design system.
- After adding the `label` and the `hint` (the text) to the `Radio` component, the old `RadioGroup` didn't need to show text or style anything anymore (why most of this diff is gone).
- Radio buttons are generally easy to group together so having a presentational grouping component seems redundant.
- Keeping the `RadioGroup` in the editor keeps the `useRadioNavigation` specific to the editor.

---

**Note** The only benefit I see for adding a radio group to the design system would be to get accessibility baked into a group of radio buttons. Like above it is not difficult to make radio buttons accessible, but we'd have to remember to:
- add a `div` with `role="radiogroup"`
- add text that describes the radio group and connect it to the radiogroup using `aria-labelledby`

If we want a `RadioGroup` component in the design system I can add it. It would look like this:

```js
function RadioGroup({
  className,
  groupLabel,
  id,
  name,
  options,
  value: selectedValue,
  ...radioButtonProps
}) {
  const groupId = useMemo(() => uuidv4(), []);

  return (
    <div
      className={className}
      role="radiogroup"
      aria-labelledby={groupId}
    >
      <VisuallyHiddenLabel id={groupId} as="h4"> // This can be anything we want it to be
        {groupLabel}
      </VisuallyHiddenLabel>
      {options.map(({ helper, label, value }) => (
        <Radio
          key={value}
          name={name}
          value={value}
          checked={value === selectedValue}
          label={label}
          hint={helper}
          {...radioButtonProps}
        />
      ))}
    </div>
};
```

Reply:
I'm good with this decision. The only reason we need the more elaborate radio group in the editor is that we have to override our own global keyboard listeners and (slightly annoyingly) recreate browser default behavior.